toc/working-groups/contributions-for-user.sh

```diff
@@ -0,0 +1,122 @@
+    echo "${commits}" |  jq -r '.[] | "- " + .commit.committer.date + ": [" + (.commit.message | split("\n"))[0] + "](" + .html_url + ")"'
+  fi
+}
+
+find_prs_for_repo() {
+  local user=$1
+  local repo=$2
+  
+  echo "${EVENTS_JSON}" | jq -r --arg repo "${repo}" '.[] | select([.type] | inside(["PullRequestReviewEvent"])) | select(.repo.name == $repo) | "- " + .payload.pull_request.created_at + ": [" +.payload.pull_request.title + "](" + .payload.pull_request.html_url + ")"'
+  echo "${EVENTS_JSON}" | jq -r --arg user "${user}" --arg repo "${repo}" '.[] | select([.type] | inside(["PullRequestReviewCommentEvent"])) | select(.repo.name == $repo) | select(.payload.pull_request.user.login != $user) | "- " + .payload.pull_request.created_at + ": [" +.payload.pull_request.title + "](" + .payload.pull_request.html_url + ")"'
+  
+}
+
+find_issues_for_repo() {
+  local user=$1
+  local repo=$2
+  
+  if issues=$(gh api --paginate "repos/${repo}/issues?creator=${user}&state=all" 2>/dev/null); then
+    echo "${issues}" | jq -r '.[] | "- " + .created_at + ": [" + .title + "](" + .html_url + ")"'
+  fi
+  if issues=$(gh api --paginate "repos/${repo}/issues?mentioned=${user}&state=all" 2>/dev/null); then
+    echo "${issues}" | jq -r '.[] | "- " + .created_at + ": [" + .title + "](" + .html_url + ")"'
+  fi
+  echo "${EVENTS_JSON}" | jq -r --arg user "${user}" --arg repo "${repo}" '.[] | select([.type] | inside(["IssueCommentEvent"])) | select(.repo.name == $repo) | select(.payload.issue.user.login != $user) | "- " + .payload.issue.created_at + ": [" +.payload.issue.title + "](" + .payload.issue.html_url + ")"'
+}
+
+  usage_checks
+  set -u
+
+  echo "Gathering GH events..."
+  EVENTS_JSON=$(gh api --paginate "users/${USERNAME}/events?per_page=100")
+
+  declare -a areas
+  areas=($(echo "${TOC_JSON}" | jq --arg wg_name "${WORKING_GROUP}" -r '.[] | select(.name == $wg_name) | .areas[].name'))
+  for area in "${areas[@]}"; do
+    echo "Gathering contributions for the '${WORKING_GROUP}: ${area}' area..."
+    gist_url=$( (
+    echo "# ${WORKING_GROUP}: ${area} Contributions"
+    declare -a repos
+    repos=($(echo "${TOC_JSON}" | jq -r --arg wg_name "${WORKING_GROUP}" --arg area "${area}" '.[] | select(.name == $wg_name) | .areas[] | select(.name == $area) | .repositories[]'))
+    echo "### PRs Commented on/Reviewed:"
+    for repo in "${repos[@]}"; do
+      find_prs_for_repo "${USERNAME}" "${repo}"
+    done | sort -u
+    echo "### Issues that may be relevant:"
+    for repo in "${repos[@]}"; do
+      find_issues_for_repo  "${USERNAME}" "${repo}"
+    done | sort -u
+    echo "### Code contributions:"
+    for repo in "${repos[@]}"; do
+        find_commits_for_repo "${USERNAME}" "${repo}" 
+    done | sort -u
+    echo
+    ) | gh gist create -d "${USERNAME}'s Possible Contributions to ${WORKING_GROUP} - ${area}" -f "${wg} - ${area}.md" - 2>/dev/null)
```

Comment:
The generation of gist for a WG area with a `/` in it name fails. E.g. in the Foundational Infrastructure WG we have the area `Integrated Databases (Mysql / Postgres)`. Trying to generate a gist for that area fails with:
```
- Creating gist Foundational Infrastructure - Integrated Databases (Mysql \/ Postgres).md
X Failed to create gist: HTTP 422: Validation Failed (https://api.github.com/gists)
Gist.files is invalid
```
You can try this out with the following command:
`echo "test" | gh gist create -d '<replace-with-a-user>'\''s Possible Contributions to Foundational Infrastructure - Integrated Databases (Mysql / Postgres)' -f 'Foundational Infrastructure - Integrated Databases (Mysql - Postgres).md' -`

Reply:
@beyhan I just ran your gist create command successfully:

```$ echo "test" | gh gist create -d '<replace-with-a-user>'\''s Possible Contributions to Foundational Infrastructure - Integrated Databases (Mysql / Postgres)' -f 'Foundational Infrastructure - Integrated Databases (Mysql - Postgres).md' -
- Creating gist Foundational Infrastructure - Integrated Databases (Mysql - Postgres).md
✓ Created gist Foundational Infrastructure - Integrated Databases (Mysql - Postgres).md
https://gist.github.com/geofffranks/9a8502331d6148c8f80de1bb3ae0e9ef
```
GH CLI:
```
gh version 2.4.0+dfsg1 (2022-03-23 Ubuntu 2.4.0+dfsg1-2)
```

Is it still failing for you? 422 might also indicate it thought there was [endoint spamming going on](https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#create-a-gist--status-codes). Though i'd expect a better error than "Gist.files is invalid" for that.

Reply:
Fixed!

```
$ ./toc/working-groups/contributions-for-user.sh beyhan "Foundational Infrastructure"
Parsing working group metadata...
Gathering GH events...
Gathering contributions for the 'Foundational Infrastructure: Credential Management (Credhub)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/44d68af48c6bfa3a3d21bd3d2a2c745e
Gathering contributions for the 'Foundational Infrastructure: Disaster Recovery (BBR)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/8e50ae1e0ada552f50256a866cbe2590
Gathering contributions for the 'Foundational Infrastructure: Identity and Auth (UAA)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/43fc2b40a5d60e47f46141b5f93bba30
Gathering contributions for the 'Foundational Infrastructure: Integrated Databases (Mysql / Postgres)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/60fd2ab2cf4d0da7affe4c1c54eaf5b7
Gathering contributions for the 'Foundational Infrastructure: System Logging and Metrics (rsyslog / event-log)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/fd3a344080e97d0377d99fe9d80b6f83
Gathering contributions for the 'Foundational Infrastructure: Stemcell Release Engineering (BOSH)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/c7b091835af211725656ebba9182f1a0
Gathering contributions for the 'Foundational Infrastructure: VM deployment lifecycle (BOSH)' area...
Contributions have been summarized at https://gist.github.com/geofffranks/62d6ea7c650bf9755e5aaa31302c6a68
```

Reply:
Thank you!